driver-core/src/main/com/mongodb/MongoClientSettings.java

```diff
@@ -820,6 +820,144 @@ public ContextProvider getContextProvider() {
         return contextProvider;
     }
 
+
+    @Override
+    public boolean equals(final Object o) {
```

Comment:
Huh, I don't know why IntelliJ isn't giving me the option to generate that style.  

Any idea why it's not using Objects.equals for uuidRepresentation? I'm guessing for efficiency, similar to boolean fields above.

Done.

Reply:
> why IntelliJ isn't giving me the option to generate that style.

Do you have the "Template" drop-down list in the "Generate equals() and hashCode()" dialog (https://www.jetbrains.com/help/idea/generating-code.html#generate-equals-hashcode)? In this list there is a template called "... (java 7+)", that's the one I am using.

> not using Objects.equals for uuidRepresentation? I'm guessing for efficiency

Yes, it is always fine to compare `Enum` objects by reference.

Similarly, `Object.equals` does less actions and has less branching than `Object`**s**`.equals`, so at the very least while being interpreted, the former should perform better than the latter. The wizard also asks to check non-null reference fields, and uses `Object.equals` to compare them. But since we don't have fields annotated with `@Nullable`, I went with not checking any as non-null, which is why `Objects.equals` is used.

Reply:
I see it now.  I just overlooked it.